ibis/impala/udf.py

```diff
@@ -336,6 +341,31 @@ def _ibis_string_to_impala(tval):
         return repr(result)
 
 
+_COMPLEX_RE = re.compile(r'(struct|map|array)\<([\s\S]+)\>$')
+_STRUCT_FIELD_RE = re.compile(
+    r"`?(\w+)`?:((?:struct|map|array)\<[\s\S]+?\>|\S+)(?:\s+comment '([^']+)')?(?:,\n|$)") # noqa E501
+_MAP_FIELD_RE = re.compile(r"([^,]+),((?:struct|map|array)\<[\s\S]+?\>|.*)")
+
+
+def _impala_complex_type_to_ibis(tval):
+    t_type, t_schema = _COMPLEX_RE.findall(tval)[0]
```

Comment:
Is `findall` the right method to use here? What if it returns multiple matches? Is that possible?

Reply:
There should only ever be one match but I have changed this to `.match(tval).groups()`